credential-management/non-fully-active.https.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<meta charset="utf-8" />
+<title>Credential Management API Test: non-fully active document</title>
+<link
+  rel="help"
+  href="https://github.com/w3c/webappsec-credential-management"
+/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<body></body>
+<script>
+  promise_test(async (t) => {
+    // Create the iframe, wait for it to load...
+    const iframe = document.createElement("iframe");
+
+    // ...wait for the iframe to load...
+    await new Promise((resolve) => {
+      iframe.addEventListener("load", resolve, { once: true });
+      iframe.src = "resources/iframe.html";
+      document.body.appendChild(iframe);
+    });
+
+    // Steal credentials container.
+    const { credentials } = iframe.contentWindow.navigator;
+
+    // No longer fully active.
+    iframe.remove();
+
+    // Try to get credentials while not fully active...
+    await promise_rejects_dom(
+      t,
+      "NotAllowedError",
+      credentials.get({ password: true }),
```

Comment:
One problem we have is that password credentials haven't been implemented by Firefox and Safari, and this tests depends on them being implemented. Otherwise, this promise will always reject.

Should we perhaps try a simple webauthn assertion instead, with a comment explaining what we're doing? I never thought about this problem, so I'm open to ideas.

Reply:
So, I think this should be ok because the credential type support check happens _after_ the fully active check:

1. Let settings be the current settings object.
2. Assert: settings is a secure context.
3. If options.signal is aborted, then return a promise rejected with options.signal’s abort reason.
4. If settings’s relevant global object's associated Document is not fully active, then return a promise rejected. 

Oh! I think I just realized there might be another bug in the spec.

Reply:
To be sure... this should work just as well:

```
credentials.get({  })
```

Reply:
Sounds good -- I just checked that finding an unknown credential type throws NotSupportedError instead of NotAllowedError, so we should be good with either.